### PR TITLE
rar2fs: init at 1.28.0

### PR DESCRIPTION
**Pull request by kraem**
###### Motivation for this change
`rar2fs` uses fuse to mount rar archives, or directories containing rar archives, and makes the files available as if they were unrar:ed.

[rar2fs Github](https://github.com/hasse69/rar2fs)
[rar2fs web](https://hasse69.github.io/rar2fs/)

###### Things done
Init package at most recent version 1.28.0

Added the unfree license even though `rar2fs` is licensed using GPLv3.
The source from `RARLAB` is used during compilation though. The license of `RARLAB` is free to distribute if only the `UnRAR` sources are used ([source](https://fedoraproject.org/wiki/Licensing:Unrar)). I'm not sure if anything else is used during the compilation of `rar2fs`. I'm super new to licenses and if anyone has any idea as to what the situation is here they are very welcome to have a say in this :)

###### Other
This is my first package added so added myself to the maintainers-list. I'm more than happy to receive suggestions for improvement too :)

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by kraem** [comment]
Woops. Thought I finally understood git history. Apparently not.

File: pkgs/tools/filesystems/rar2fs/default.nix
Comment (primeos) [inline pkgs/tools/filesystems/rar2fs/default.nix]: You can use `stdenv.mkDerivation rec {` here (`rec` being the important change) and move `version` and `unrar-src` into the set below to avoid the `let ... in` (since you inherit both attributes anyway).

File: pkgs/tools/filesystems/rar2fs/default.nix
Comment (primeos) [inline pkgs/tools/filesystems/rar2fs/default.nix]: Can't you depend on the existing `unrar` package (`pkgs/tools/archivers/unrar/default.nix`) instead of fetching and building it here?

Let me know if it causes problems (e.g. if the build scripts don't find the system-wide library).

File: pkgs/tools/filesystems/rar2fs/default.nix
Comment (primeos) [inline pkgs/tools/filesystems/rar2fs/default.nix]: `fetchFromGitHub` is better in this case, though both variants will do more or less the same thing ;)

File: pkgs/tools/filesystems/rar2fs/default.nix
Comment (primeos) [inline pkgs/tools/filesystems/rar2fs/default.nix]: You can use `gpl3Plus` here if you depend on `unrar` ;)

**Review by primeos — CHANGES_REQUESTED** [review]
Hi @kraem, welcome to Nixpkgs and thanks for your contribution :)

I found a few things that should be improved, but overall it looks good.

File: pkgs/tools/filesystems/rar2fs/default.nix
Comment (kraem) [reply pkgs/tools/filesystems/rar2fs/default.nix]: Thanks! Will do.

File: pkgs/tools/filesystems/rar2fs/default.nix
Comment (kraem) [reply pkgs/tools/filesystems/rar2fs/default.nix]: I tried that but the build scripts wants the source while compiling rar2fs. Will try again :)

File: pkgs/tools/filesystems/rar2fs/default.nix
Comment (kraem) [reply pkgs/tools/filesystems/rar2fs/default.nix]: I'll fix!

File: pkgs/tools/filesystems/rar2fs/default.nix
Comment (kraem) [reply pkgs/tools/filesystems/rar2fs/default.nix]: Since the build script is dependant on **unrarsrc**-5.8.5.tar.gz from https://www.rarlab.com/rar/, that makes sense :)

File: pkgs/tools/filesystems/rar2fs/default.nix
Comment (primeos) [reply pkgs/tools/filesystems/rar2fs/default.nix]: If the source is needed it should work with something like "--with-unrar=${unrar.src}" (if the store path is already extracted). I also had a look at [configure.ac](https://github.com/hasse69/rar2fs/blob/b6d1d42952d8ec7153bc1c859de88b288cf6b5d6/configure.ac#L209) and `--disable-static-unrar` probably makes sense as well as the default is to link statically (that might be why the source-code is required).

File: pkgs/tools/filesystems/rar2fs/default.nix
Comment (kraem) [reply pkgs/tools/filesystems/rar2fs/default.nix]: As soon as I introduce
```
  src = fetchFromGitHub {
    owner = "hasse69";
    repo = "rar2fs";
    rev = "b6d1d42952d8ec7153bc1c859de88b288cf6b5d6";
    sha256 = "0fmdqrs5yvn89ngc26vj5ggnalpwrdm8pdcfszw1wflh78hvd8kb";
  };
```
I get this error: `error: some outputs of '/nix/store/8zbcmjca0nr0r68jgpmay2z599hlx9fm-rar2fs-1.28.0.drv' are not valid, so checking is not possible` and I can't figure out why.

File: pkgs/tools/filesystems/rar2fs/default.nix
Comment (kraem) [reply pkgs/tools/filesystems/rar2fs/default.nix]: Also here I get
```
these derivations will be built:
  /nix/store/h0v6fvccw445zv7n4adbjl48f8fk7bwd-rar2fs-1.28.0.drv
error: some outputs of '/nix/store/h0v6fvccw445zv7n4adbjl48f8fk7bwd-rar2fs-1.28.0.drv' are not valid, so checking is not possible
```
when introducing
```
  configureFlags = [
    "--disable-static-unrar"
    "--with-unrar=${unrar.src}"
  ];
```
incombination `buildInputs = [ fuse unrar ];` and `{ unrar }` in the arguments in the top of the nix-file.

File: pkgs/tools/filesystems/rar2fs/default.nix
Comment (primeos) [reply pkgs/tools/filesystems/rar2fs/default.nix]: Yes, this was assuming that we can re-use `unrar` from Nixpkgs which already has the correct license (and if that wouldn't be possible this could also be a list like `with licenses; [ gpl3Plus unfree]` - but in this case we should just depend on `unrar`).

File: pkgs/tools/filesystems/rar2fs/default.nix
Comment (kraem) [reply pkgs/tools/filesystems/rar2fs/default.nix]: When introducing 
```
  src = fetchFromGitHub {
    owner = "hasse69";
    repo = "rar2fs";
    rev = "b6d1d42952d8ec7153bc1c859de88b288cf6b5d6";
    sha256 = "0fmdqrs5yvn89ngc26vj5ggnalpwrdm8pdcfszw1wflh78hvd8kb";
  };
```
 it seems the working dir is `/build/source` instead of `/build/rar2fs-1.28.0` and it can't find the configure script
```
configuring
no configure script, doing nothing
```
as opposed to
```
configuring
configure flags: --disable-static --disable-dependency-tracking --prefix=/nix/store/wbwmdlmqys0gv311iylwifjq6wd5wiy0-rar2fs-1.28.0 --with-unrar=/build/unrar/ --with-unrar-lib=/build/unrar/
```

Any thoughts?

Comment (primeos) [reply pkgs/tools/filesystems/rar2fs/default.nix]: Strange, this should normally work. For special-cases there is `sourceRoot` (https://nixos.org/nixpkgs/manual/#ssec-unpack-phase) but that shouldn't be required in this case (from a quick look at the GH repo).

Btw: You should use `rev = "v${version}` instead of the commit hash ;)

Also, it could be that `0fmdqrs5yvn89ngc26vj5ggnalpwrdm8pdcfszw1wflh78hvd8kb` is a wrong or old hash (you can use `nix-prefetch-url -A rar2fs.src` to get the correct one).

If it doesn't work feel free to push the current version, then I can have a look what's going on.

Comment (kraem) [reply pkgs/tools/filesystems/rar2fs/default.nix]: Ok so I've figured out that tools like `automake`, `autoconf` and other utilities like `which` (which is needed by `get-version.sh`, no pun intended ;) ) are not readily available when using fetchFromGitHub instead of fetchurl so the configure phase does not succeed.

Is there any easy fix to this?

Comment (kraem) [reply pkgs/tools/filesystems/rar2fs/default.nix]: Ok so I've found `autoreconfHook` which seems to do the magic. :)

Comment (kraem) [reply pkgs/tools/filesystems/rar2fs/default.nix]: Sorry didn't see your last reply. I did figure the `rev = "v${version}"` out after some thinking :)
Thanks for the tip on with `nix-prefetch-url -A` instead of `nix-prefetch-url <URL>`

**Comment by primeos** [comment]
Finally managed to build it without the unrar source-code, wasn't used to debugging autoconf scripts anymore.. :o :D
Currently testing the rebuilds with `nix-review` but from a quick look at Nixpkgs it should be fine (at least mostly).

@kraem could you test if everything still works as expected?

cc: @ehmry as the `unrar` maintainer

**Comment by kraem** [comment]
@primeos Works perfectly. Thanks!

**Review by primeos — APPROVED** [review]
LGTM now, thanks for the PR :)

@ehmry I'll merge it tomorrow if you don't have any complaints (just in case I missed something) ;)

Edit: Forgot to mention: `nix-review` was successful:
```
6 package were build:
pdfread rar2fs rarcrack sabnzbd unrar zod
```

**Pull request merged**
